include/envoy/common/platform.h

```diff
@@ -19,13 +19,24 @@
 #include <mswsock.h>
 #include <ws2tcpip.h>
 
+#undef NTDDI_VERSION
+#define NTDDI_VERSION WDK_NTDDI_VERSION
+
+#include <mstcpip.h>
+
+#ifndef SIO_TCP_INFO
+  static_assert(false, "SIO_TCP_INFO not defined");
```

Comment:
@wrowe  this is the hack I was talking about offline. If you comment out line 22 and 23 then this static assert hits